components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -187,6 +229,10 @@ func (sb *sensorBase) isPolling() bool {
 func (sb *sensorBase) MoveStraight(
 	ctx context.Context, distanceMm int, mmPerSec float64, extra map[string]interface{},
 ) error {
+	if sb.loop != nil {
+		sb.loop.Stop()
+		sb.loop = nil
+	}
```

Comment:
@randhid I think this did it, just spammed it with many different calls for a while and haven't seen the errors again